esphome/components/gps/__init__.py

```diff
@@ -1,7 +1,22 @@
 import esphome.codegen as cg
 import esphome.config_validation as cv
 from esphome.components import uart
-from esphome.const import CONF_ID
+from esphome.components import sensor
+from esphome.const import (
+    CONF_ID,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_SPEED,
+    CONF_COURSE,
+    CONF_ALTITUDE,
+    CONF_SATELLITES,
+    UNIT_DEGREES,
+    UNIT_KILOMETER_PER_HOUR,
+    UNIT_METER,
+    UNIT_EMPTY,
+    ICON_EMPTY,
+    DEVICE_CLASS_EMPTY,
+)
 
 DEPENDENCIES = ["uart"]
```

Comment:
```suggestion
DEPENDENCIES = ["uart"]
AUTO_LOAD = ["sensor"]
```

Add this so that it works even if the user doesn't have a `sensor` block declared in their YAML.

Reply:
Thanks, first time even looking at how this code works :)